python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -1590,20 +1598,21 @@ def __init__(
             raise NotImplementedError(
                 f"Conditional join with predicate {predicate}"
             )  # pragma: no cover; polars never delivers expressions we can't handle
-        self._non_child_args = (predicate_wrapper, zlice, suffix, maintain_order)
+        self._non_child_args = (predicate_wrapper, options)
```

Comment:
This change, and the change to `do_evaluate`, are to align with the other IR nodes where `self.non_child_args` matches `cls._non_chlid` and the signature of `cls.do_evaluate`.

Reply:
I don't feel great about https://github.com/rapidsai/cudf/pull/19895/commits/b5c17a62432cdd7f118c9d4325617a959a44e6fe, but we have tests ensuring that `ConditionalJoin` is picklable. Perhaps that's why this was different in the first place.